File: Unique Number of Occurrences/unique-number-of-occurrences.py
class Solution(object):
    def uniqueOccurrences(self, arr):
        m = {}
        s = set()
        
        
        for num in arr:
            m[num] = m.get(num, 0) + 1
        
        
        for count in m.values():
            if count in s:
                return False
            s.add(count)
        
        return True

File: Unique Number of Occurrences/unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer, Integer> m = new HashMap<>();
        HashSet<Integer> s = new HashSet<>();

        for (int num : arr) {
            m.put(num, m.getOrDefault(num, 0) + 1);
        }

        for (int count : m.values()) {
            if (s.contains(count)) {
                return false;
            }
            s.add(count);
        }

        return true;
    }
}
